bokeh/charts/_charts.py

```diff
@@ -94,109 +84,103 @@ def __init__(self, title, xlabel, ylabel, legend, xscale, yscale, width, height,
                 categorical plot.
             glyphs (list): to keep track of the glyphs added to the plot.
         """
-        self.title = title
-        self.xlabel = xlabel
-        self.ylabel = ylabel
-        self.legend = legend
-        self.xscale = xscale
-        self.yscale = yscale
-        self.plot_width = width
-        self.plot_height = height
-        self.tools = tools
-        self.filename = filename
-        self.server = server
-        self.notebook = notebook
-        self._xdr = None
-        self._ydr = None
-        self.facet = facet
-        self._plots = []
-        self.categorical = False
-        self.glyphs = []
+        kw = dict(title=title, plot_width=width, plot_height=height)
+
+        self._source = None
+        # list to save all the groups available in the incomming input
+        self._groups = []
+        self._data = dict()
+        self._attr = []
+
+        super(Chart, self).__init__(**kw)
+
+        self.__title = title
+        self.__xlabel = xlabel
+        self.__ylabel = ylabel
+        self.__legend = legend
+        self.__xscale = xscale
+        self.__yscale = yscale
+        self.__width = width
+        self.__height = height
+        self.__enabled_tools = tools
+        self.__filename = filename
+        self.__server = server
+        self.__notebook = notebook
+        self.__facet = facet
+        self.__palette = palette
+        self.__xgrid = xgrid
+        self.__ygrid = ygrid
+
+        self._glyphs = []
+        self._built = False
+
+        self._builders = []
+        self._renderer_map = []
 
         # Add to document and session if server output is asked
-        if doc:
-            self.doc = doc
-            if not self.doc._current_plot:
-                self.figure()
-            else:
-                self._plots = [self.doc._current_plot]
+        if _doc:
+            self._doc = _doc
         else:
-            self.figure()
-            self.doc = Document()
+            self._doc = Document()
 
-        if self.server:
-            if session:
-                self.session = session
+        if self.__server:
+            if _session:
+                self._session = _session
             else:
-                self.session = Session()
+                self._session = Session()
 
-    @property
-    def plot(self):
-        """
-        Returns the currently chart plot
-        """
-        return self._plots[-1]
+        self.check_attr()
```

Comment:
until we solve this in a better way a simple comment about the need to check the attr would be good...